doc/mgmt_preview_quickstart.md

```diff
@@ -0,0 +1,338 @@
+            };
+
+            ipAddress = await publicIpAddressClient.StartCreateOrUpdate(resourceGroupName, vmName + "_ip", ipAddress)
+                .WaitForCompletionAsync();
+
+            // Create VNet
+            var vnet = new VirtualNetwork()
+            {
+                Location = location,
+                AddressSpace = new AddressSpace() { AddressPrefixes = new List<string>() { "10.0.0.0/16" } },
+                Subnets = new List<Subnet>()
+                {
+                    new Subnet()
+                    {
+                        Name = "mySubnet",
+                        AddressPrefix = "10.0.0.0/24",
+                    }
+                },
+            };
+
+            vnet = await virtualNetworksClient
+                .StartCreateOrUpdate(resourceGroupName, vmName + "_vent", vnet)
+                .WaitForCompletionAsync();
+
+            // Create Network interface
+            var nic = new NetworkInterface()
+            {
+                Location = location,
+                IpConfigurations = new List<NetworkInterfaceIPConfiguration>()
+                {
+                    new NetworkInterfaceIPConfiguration()
+                    {
+                        Name = "Primary",
+                        Primary = true,
+                        Subnet = new Subnet() { Id = vnet.Subnets.First().Id },
+                        PrivateIPAllocationMethod = IPAllocationMethod.Dynamic,
+                        PublicIPAddress = new PublicIPAddress() { Id = ipAddress.Id }
+                    }
+                }
+            };
+
+            nic = await networkInterfaceClient
+                .StartCreateOrUpdate(resourceGroupName, vmName + "_nic", nic)
+                .WaitForCompletionAsync();
+
+            var vm = new VirtualMachine(location)
+            {
+                NetworkProfile = new Compute.Models.NetworkProfile { NetworkInterfaces = new[] { new NetworkInterfaceReference() { Id = nic.Id } } },
+                OsProfile = new OSProfile
+                {
+                    ComputerName = "testVM",
+                    AdminUsername = "username",
+                    AdminPassword = "(YourPassword)",
+                    LinuxConfiguration = new LinuxConfiguration { DisablePasswordAuthentication = false, ProvisionVMAgent = true }
+                },
+                StorageProfile = new StorageProfile()
+                {
+                    ImageReference = new ImageReference()
+                    {
+                        Offer = "UbuntuServer",
+                        Publisher = "Canonical",
+                        Sku = "18.04-LTS",
+                        Version = "latest"
+                    },
+                    DataDisks = new List<DataDisk>()
+                },
+                HardwareProfile = new HardwareProfile() { VmSize = VirtualMachineSizeTypes.StandardB1Ms },
+            };
+            vm.AvailabilitySet.Id = availabilitySet.Id;
```

Comment:
Can't directly set `vm.AvailabilitySet.Id` like that,` vm.AvailabilitySet` is null at this moment.
Can we use it as VirtualMachine's property?
`AvailabilitySet = new Azure.ResourceManager.Compute.Models.SubResource() { Id = availabilitySet.Id }`